ext/cl_khr_tensor.asciidoc

```diff
@@ -0,0 +1,584 @@
+// Copyright 2023 The Khronos Group. This work is licensed under a
+// Creative Commons Attribution 4.0 International License; see
+// http://creativecommons.org/licenses/by/4.0/
+= cl_exp_tensor
+
+:source-highlighter: coreray
+
+[[cl_exp_tensor]]
+== Tensor Data Type
+
+This extension provides a new opaque OpenCL datatype called
+`cl_tensor`. It is used for storing N-dimensional tensor data in
+implementation-defined memory layout which may be optimized based on
+tensor's use cases. The datatype is designed to be efficiently used
+within the `cl_khr_command_buffers` extension to capture task graphs
+which can utilize tensors as input, output and temporary storage.
+
+=== General information
+
```

Comment:
Let's write a motivational text here which is interfacing to the DBKs (refer to the WiP spec) on the backend, and to better interface to higher-level models for neural networks and other libraries/formats dealing with tensors.